common/check_expired_locked_rules

```diff
@@ -1,56 +1,117 @@
 
 # Exit statuses
 OK, WARNING, CRITICAL, UNKNOWN = 0, 1, 2, 3
 
+if __name__ == '__main__':
 
-def main():
-    '''
-    Probe to check the locked expired rules or datasets with locked rules
-    '''
     status = OK
-    session = get_session()
-    try:
-        query = "select rawtohex(id), scope, name, rse_expression from atlas_rucio.rules where locked=1 and expires_at<sys_extract_utc(localtimestamp)"
-        print 'Locked expired rules'
-        for row in session.execute(query):
-            status = CRITICAL
-            print row[0], row[1], row[2]
-    except Exception as error:
-        print error
-        status = UNKNOWN
-        sys.exit(status)
-    try:
-        query = """select rawtohex(c.id), c.scope, c.name, c.rse_expression from atlas_rucio.rules c,
-                   (select a.scope, a.name from atlas_rucio.dids a
-                   where a.expired_at is not null and a.expired_at < sys_extract_utc(localtimestamp)
-                   and exists (select 1 from atlas_rucio.rules b where a.scope=b.scope and a.name=b.name and locked=1)) d
-                   where c.scope=d.scope and c.name=d.name and locked=1"""
-        print 'Datasets expired with locked rules'
-        for row in session.execute(query):
-            status = CRITICAL
-            print row[0], row[1], row[2], row[3]
-    except Exception as error:
-        print error
-        status = UNKNOWN
-        sys.exit(status)
-    sys.exit(status)
+    session = session.get_session()
+
+    # Select statement used for both metrics
+    base_statement = select(
+        models.ReplicationRule,
+        models.DataIdentifier.name,
+        models.DataIdentifier.scope,
+        models.ReplicationRule.rse_expression,
+    )
+
+    # Use prometheus pusher to send results to a remote service
+    with PrometheusPusher() as manager:
+        try:
+            statement = base_statement.where(
+                and_(
+                    models.ReplicationRule.locked == true(),
+                    models.ReplicationRule.expires_at < datetime.utcnow()  # see https://github.com/rucio/rucio/issues/6476
+                )
+            )
+            rule_counts = defaultdict(int)
 
+            # Print rules for nagios monitoring
+            print("Locked expired rules")
+            for row in session.execute(statement):
+                print(row.rule_id, row.scope, row.name, row.rse_expression)
+                status = CRITICAL
+                # Keep track of the counts
+                rule_counts[row.rse_expression] += 1
 
-if __name__ == "__main__":
-    main()
+            # Add a summary entry so when there are no result from the query there is metric continuity
+            rule_counts["All"] = sum(rule_counts.values())
+
+            # Send to Prometheus pusher
+            for rse_expression, count in rule_counts.items():
+                (manager.gauge(
+                        "locked_expired_rules.{rse_expression}",
```

Comment:
I guess this is supposed to be an f-string?

Reply:
No, Prometheus reformats strings server side using the {} as a guide to insert labels. - [see this example in CMS](https://github.com/rucio/probes/blob/master/cms/check_rules_states_by_account#L67)